pkg/schema/schema.go

```diff
@@ -129,11 +137,17 @@ func NewArrayItemType(item *yamlmeta.ArrayItem) (*ArrayItemType, error) {
 		return nil, err
 	}
 
-	annotations := template.NewAnnotations(item)
+	templateAnnotations := template.NewAnnotations(item)
 
-	if _, found := annotations[AnnotationSchemaNullable]; found {
+	if _, found := templateAnnotations[AnnotationSchemaNullable]; found {
 		return nil, NewInvalidSchemaError(item, fmt.Sprintf("@%s is not supported on array items", AnnotationSchemaNullable), "")
 	}
+	anyType, err := processTypeAnnotation(templateAnnotations)
```

Comment:
Can we consolidate the ways we check for the "various" (😁  there's two right now) annotations? Not sure if it fits, but I have an impulse to use a `switch` here, for example.

Reply:
with `schema/nullable` and `schema/type`, is it possible to annotate a node with both annotations? If not, I agree that a switch statement would be the way to go.

Reply:
is this a valid schema definition?
```
#@schema/match data_values=True
---
#@schema/type any=True
foo: nil
```
just as this is a valid schama:
```
#@schema/match data_values=True
---
#@schema/nullable
foo: nil
```

Reply:
This may have already been resolved offline, but my take here is yes this is valid

Reply:
+1 @cari-lynn.

We _did_ talk about it offline. Meant to summarize here. Will do now.

That:
- "null" is the one value of the type "null" and so it is a legitimate value to have in a schema
- `type any=True` implies _all_ types, which includes "null" type

And:
- while `@schema/nullable` over a "null" is technically valid, it's also useless (a variable that can't vary?) but not seeing a need to attempt to prevent it, right now.
- one use-case of an author using `type any=True` is a required key whose contents are merely passed along: "null" seems like a reasonable default, here.

Did I miss anything @gcheadle-vmware ?